### PR TITLE
@observable decorator

### LINKED ISSUE #785: Support for `@observable` inside of view-models
# 🙋 Feature Request

Add in support for the `@observable` functionality for observing changes to class variables inside of a view-model.

## 🤔 Expected Behavior

Users should be able to import the `@observable` decorator as per v1 and observe changes via change handlers on one or more properties inside of a class.

## 😯 Current Behavior

Not implemented.

### PR DESCRIPTION
**Pull request by bigopon**
# Pull Request

## 📖 Description

- Clean up lifecycle dependency in setter observer.
- Port `@observable` decorator

### 🎫 Issues

Closes #785 

## ⏭ Next Steps

Continue with cleaning up observation, preparing for next static observation.

### CODE REVIEW

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54631-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54702-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54761-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

File: packages/runtime/src/observation/observable.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      descriptor = { configurable: true };
+    }
+
+    // make the accessor enumerable by default, as fields are enumerable
+    if (!('enumerable' in descriptor)) {
+      descriptor.enumerable = true;
+    }
+
+    // todo(bigopon/fred): discuss string api for converter
+    const $set = config.set;
```
Comment (bigopon) [inline packages/runtime/src/observation/observable.ts:126]: A missing feature is ability to read TS metadata to figure out the converter. Adding this should gives some ability to make things more ergonomic as well.
For v2, when we do:
```ts
import { IPoint } from 'my-point';

class App {
  @observable
  point: IPoint
}
```
With TS, metadata generated will have `IPoint` for property `point`. And if `IPoint` was created with `DI.createInterface`, then the `@observable` decorator can auto look for a `set` property on this interface "object" to determine how to convert incoming value:
```ts
export IPoint = DI.createInterface();
IPoint.set = function(pointString) {
  return pointString.split(' ');
}
```

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://54881-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Pull request merged**